test/functional/mempool_compatibility.py

```diff
@@ -37,25 +33,26 @@ def setup_network(self):
             None,
         ])
         self.start_nodes()
-        self.import_deterministic_coinbase_privkeys()
 
     def run_test(self):
         self.log.info("Test that mempool.dat is compatible between versions")
 
         old_node, new_node = self.nodes
-        new_wallet = MiniWallet(new_node, mode=MiniWalletMode.RAW_P2PK)
+        new_wallet = MiniWallet(new_node, mode=MiniWalletMode.RAW_P2PK_RANDOM)
+        old_wallet = MiniWallet(old_node, mode=MiniWalletMode.RAW_P2PK_RANDOM)
```

Comment:
Generally MiniWallet is designed to use the latest features of Bitcoin Core, so it may not work with older versions. Also, it should be sufficient to have a single wallet instance per test.
Thus, it should be possible to remove the below `generatetodescriptor` and just replace `1` blocks with `2` in the line below. And also the "random" miniwallet mode is not needed.

Reply:
Done in https://github.com/bitcoin/bitcoin/pull/26640 , which should hopefully allow to drop two commits from this pull and make it simpler.